reader.go

```diff
@@ -31,17 +33,28 @@ func Decode(r io.Reader) (image.Image, error) {
 		read += 4
 		switch string(next) {
 		case "TOC ":
-			read += 4
+			resSize := binary.BigEndian.Uint32(data[read : read+4])
+			read += resSize-4 // size includes header and size fields
```

Comment:
Given that `resSize` here specifically refers to the "size of the TOC data", it might be good idea to name accordingly - something like `tocSize` or `tocLength`.

This is primarily because it has no relation to `resSize` further down, but it shares the same name, which is confusing.

Reply:
Sure thing, fixed